### PR TITLE
ternimal: init at 0.1.0-unstable-2017-12-31

### PR DESCRIPTION
**Pull request by Anomalocaridid**
## Description of changes

Adds [ternimal](https://github.com/p-e-w/ternimal), a program that renders an animated creature in a terminal.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
@Anomalocaridid I am not this project is a good fit to be added to nixpkgs

mainly because its just a few commits made 7 years ago

but if its packaged and it works, maybe you can still share the derivation on https://discourse.nixos.org/

**Comment by wegank** [comment]
I'm fine with adding the package as long as it works with the latest version of rustc.

(but I won't click the merge button until everyone agrees)

File: pkgs/by-name/te/ternimal/package.nix
Comment (wegank) [inline pkgs/by-name/te/ternimal/package.nix:8]:
```suggestion
  version = "0.1.0-unstable-2017-12-31";
```

https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming

File: pkgs/by-name/te/ternimal/package.nix
Comment (wegank) [inline pkgs/by-name/te/ternimal/package.nix:12]:
```suggestion
    repo = "ternimal";
```

Technically, it is unrelated to `pname`, which is rather arbitrary.

File: pkgs/by-name/te/ternimal/package.nix
Comment (wegank) [inline pkgs/by-name/te/ternimal/package.nix:32]: `meta.platforms` isn't set.

https://github.com/NixOS/nixpkgs/tree/master/pkgs#new-packages

File: pkgs/by-name/te/ternimal/package.nix
Comment (kirillrdy) [reply pkgs/by-name/te/ternimal/package.nix:12]: more info here https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/te/ternimal/package.nix
Comment (kirillrdy) [inline pkgs/by-name/te/ternimal/package.nix:19]: this should be split into buildPhase and installPhase, i know may seem like an overkill for something this simple

File: pkgs/by-name/te/ternimal/package.nix
Comment (kirillrdy) [inline pkgs/by-name/te/ternimal/package.nix:36]: sorry for adhoc feedback, i should have spotted it earlier

`mainProgram` should be set 

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

Comment (Anomalocaridid) [reply pkgs/by-name/te/ternimal/package.nix:36]: No worries. It happens.

**Review by kirillrdy — COMMENTED** [review]
image

Result of `nixpkgs-review pr 301145` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ternimal</li>
  </ul>
</details>

**Comment by Anomalocaridid** [comment]
Is there anything else I need to do before this can be merged?

**Review by kirillrdy — APPROVED** [review]

**Comment by Anomalocaridid** [comment]
Thank you!

**Pull request merged**
